lib/index.js

```diff
@@ -147,9 +147,20 @@ api.set = async ({
 
     const {ttl} = cfg.nonce.defaults;
 
+    // check for existing salt and generate new one if not found
+    let salt;
+    try {
+      const token = await api.get({actor: null, account, email, type});
```

Comment:
We already have `tokens` from the above `getAll` call that we should use instead of doing another database hit.

If `tokens.length > 0`, we should get its salt and parse its number of rounds. If the number of rounds matches the config, it's reusable. If not, we should act as if there is no salt (we need to generate one). This gives us upgradability.